hawkular-inventory-api/src/main/java/org/hawkular/inventory/api/model/DataEntity.java

```diff
@@ -0,0 +1,165 @@
+        this.value = value;
+    }
+
+    public DataEntity(CanonicalPath owner, Role role, StructuredData value, Map<String, Object> properties) {
+        super(owner.extend(DataEntity.class, role.name()).get(), properties);
+        this.value = value;
+    }
+
+    public DataEntity(CanonicalPath path, StructuredData value, Map<String, Object> properties) {
+        this(path.up(), Role.valueOf(path.getSegment().getElementId()), value, properties);
+    }
+
+    public StructuredData getValue() {
+        return value;
+    }
+
+    public Role getRole() {
+        return Role.valueOf(getId());
+    }
+
+    @Override
+    public <R, P> R accept(ElementVisitor<R, P> visitor, P parameter) {
+        return visitor.visitData(this, null);
+    }
+
+    @Override
+    public Updater<Update, ? extends AbstractElement<?, Update>> update() {
+        return new Updater<>((u) -> new DataEntity(this.getPath().up(), this.getRole(),
+                valueOrDefault(u.getValue(), getValue()), u.getProperties()));
+    }
+
+    public enum Role {
+        configuration, connectionConfiguration;
+
+        /**
+         * @return the relationship that is used to express the role of the data entity. The data entity is related
+         * to its owner by this relationship (going from the owner to the data entity).
+         */
+        public Relationships.WellKnown getCorrespondingRelationship() {
+            switch (this) {
+                case configuration:
+                    return Relationships.WellKnown.isConfiguredBy;
+                case connectionConfiguration:
+                    return Relationships.WellKnown.connectsUsing;
+                default:
+                    throw new AssertionError("Incomplete data entity role to relationship mapping.");
+            }
+        }
+    }
+
+    public static final class Blueprint extends AbstractElement.Blueprint {
+        private final StructuredData value;
+
+        public static Builder builder() {
+            return new Builder();
+        }
+
+        public Blueprint(StructuredData value, Map<String, Object> properties) {
+            super(properties);
+            this.value = value;
+        }
+
+        public StructuredData getValue() {
+            return value;
+        }
+
+        @Override
+        public <R, P> R accept(ElementBlueprintVisitor<R, P> visitor, P parameter) {
+            return visitor.visitData(this, parameter);
+        }
+
+        public static final class Builder extends Entity.Blueprint.Builder<Blueprint, Builder> {
+
+            private StructuredData value;
+
+            public Builder withValue(StructuredData value) {
+                this.value = value;
+                return this;
+            }
+
+            @Override
+            public Blueprint build() {
+                return new Blueprint(value, properties);
+            }
+        }
+    }
+
+    public static final class Update extends Entity.Update {
```

Comment:
I guess the `role` is not here and in the Blueprint.Builder, because it's inferred from the context of the api call, right? If called on `...configuration()` or `...connectConfig()`